### PR TITLE
Safety improvements (Result of the security review to v0.0.1-ac.1)

### PR DESCRIPTION
**Pull request by GNSPS**
Each commit corresponds to an issue posted on the repository earlier.

Those are also specified with the exact same wording in the report of the corresponding SemVer label.

### CODE REVIEW

File: contracts/Sale.sol [hunk 1]
```diff
@@ -42,13 +42,13 @@ contract Sale {
...
 
-    modifier saleEnded {
+    modifier afterEndBlock {
          require(block.number > endBlock);
          _;
     }
 
     modifier saleNotEnded {
-        require(block.number <= endBlock);
+        require(block.number <= endBlock && token.balanceOf(this) > 0);
```
Comment (ethers) [inline contracts/Sale.sol:51]: +1

File: contracts/Sale.sol [hunk 2]
```diff
@@ -99,6 +99,8 @@ contract Sale {
         uint _totalTimelockedBeneficiaries,
         uint _endBlock
     ) {
+        require(block.number <= _freezeBlock && _freezeBlock < _startBlock && _startBlock < _endBlock);
```
Comment (ethers) [inline contracts/Sale.sol:102]: minor note 3 expressions without parentheses less readable

File: contracts/Sale.sol [hunk 3]
```diff
@@ -150,11 +153,14 @@ contract Sale {
     )
         public
         onlyOwner
+        beforeFreeze
     {
         assert(preSaleTokensDisbursed);
         assert(!timelockedTokensDisbursed);
 
         for(uint i = 0; i < _beneficiaries.length; i++) {
+          require(owner != beneficiary);
```
Comment (ethers) [inline contracts/Sale.sol:162]: +1

**Comment by ethers** [comment]
+1 for all of these improvements (became too repetitive to add more +1)

**Comment by GNSPS** [comment]
Gonna merge it, since it's been a long time already out for review

**Pull request merged**